docs/src/blockchain-development/purity.md

```diff
@@ -2,6 +2,23 @@
 
 A function is _pure_ if it does not access any [persistent storage](./storage.md). Conversely, the function is _impure_ if it does access any storage. Naturally, as storage is only available in smart contracts, impure functions cannot be used in predicates, scripts, or libraries. A pure function cannot call an impure function.
 
-In Sway, functions are pure by default but can be opted in to impurity via function annotations which are currently work-in-progress.
+In Sway, functions are pure by default but can be opted into impurity via the `storage` function attribute.  The `storage` attribute may take `read` and/or `write` arguments indicating which type of access the function requires.
+
+```sway
+#[storage(read)]
+fn get_amount() -> u64 {
+    ...
```

Comment:
While we're here, could you file an issue to make a `[...]` annotation? Would be very useful for docs!

Reply:
I'm not sure what you mean, sorry. :slightly_smiling_face: An annotation for `mdbook` to markup `...`?

Reply:
I should have said attribute, and should have written `#[...]`. Am I imagining things or does Rust not have something like that, which allows something that looks like ellipses in documentation and such to actually be valid Rust code?

Reply:
I have no idea, not something I've seen.  Again, maybe @sezna knows..?